src/Uno.UI/Microsoft/UI/Xaml/Controls/TabView/TabView.cs

```diff
@@ -0,0 +1,1248 @@
+				{
+					ThemeShadow shadow;
+					shadow.Receivers.Add(GetShadowReceiver());
+
+					double shadowDepth = (double)SharedHelpers.FindInApplicationResources(c_tabViewShadowDepthName, c_tabShadowDepth);
+
+					var currentTranslation = shadowCaster.Translation;
+					var translation = new Vector3(currentTranslation.X, currentTranslation.Y, (float)shadowDepth);
+					shadowCaster.Translation = translation;
+
+					shadowCaster.Shadow = shadow);
+				}
+			}
+
+			UpdateListViewItemContainerTransitions();
+		}
+
+
+		private void OnListViewDraggingPropertyChanged(DependencyObject sender, DependencyProperty args)
+		{
+			UpdateListViewItemContainerTransitions();
+		}
+
+		private void OnListViewGettingFocus(object sender, GettingFocusEventArgs args)
+		{
+			// TabViewItems overlap each other by one pixel in order to get the desired visuals for the separator.
+			// This causes problems with 2d focus navigation. Because the items overlap, pressing Down or Up from a
+			// TabViewItem navigates to the overlapping item which is not desired.
+			//
+			// To resolve this issue, we detect the case where Up or Down focus navigation moves from one TabViewItem
+			// to another.
+			// How we handle it, depends on the input device.
+			// For GamePad, we want to move focus to something in the direction of movement (other than the overlapping item)
+			// For Keyboard, we cancel the focus movement.
+
+			var direction = args.Direction;
+			if (direction == FocusNavigationDirection.Up || direction == FocusNavigationDirection.Down)
+			{
+				var oldItem = args.OldFocusedElement as TabViewItem;
+				var newItem = args.NewFocusedElement as TabViewItem;
+				if (oldItem != null && newItem != null)
+				{
+					var listView = m_listView;
+					if (listView != null)
+					{
+						bool oldItemIsFromThisTabView = listView.IndexFromContainer(oldItem) != -1;
+						bool newItemIsFromThisTabView = listView.IndexFromContainer(newItem) != -1;
+						if (oldItemIsFromThisTabView && newItemIsFromThisTabView)
+						{
+							var inputDevice = args.InputDevice;
+							if (inputDevice == FocusInputDeviceKind.GameController)
+							{
+								var listViewBoundsLocal = new Rect(0, 0, (float)listView.ActualWidth, (float)listView.ActualHeight);
+								var listViewBounds = listView.TransformToVisual(null).TransformBounds(listViewBoundsLocal);
+								FindNextElementOptions options = new FindNextElementOptions();
+								options.ExclusionRect = listViewBounds;
+								var next = FocusManager.FindNextElement(direction, options);
+								var args2 = args as GettingFocusEventArgs; //TODO:MZ: IGettingFocusEventArgs2?
+								if (args != null)
+								{
+									args2.TrySetNewFocusedElement(next);
+								}
+
+								else
+								{
+									// Without TrySetNewFocusedElement, we cannot set focus while it is changing.
+									m_dispatcherHelper.RunAsync([next]()
+		
+									{
+										SetFocus(next, FocusState.Programmatic);
+									});
```

Comment:
Codacy found an issue: [Remove this empty statement.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6063676)